src/main/java/org/takes/facets/auth/PsBasic.java

```diff
@@ -0,0 +1,183 @@
+import org.takes.misc.Opt;
+
+/**
+ * Pass that checks the user according RFC-2617.
+ *
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Endrigo Antonini (teamed@endrigo.com.br)
+ * @version $Id$
+ * @since 0.20
+ */
+@EqualsAndHashCode(of = { "entry" })
+public final class PsBasic implements Pass {
+
+    /**
+     * Authorization response HTTP head.
+     */
+    private static final String AUTH_HEAD = "Authorization: Basic";
+
+    /**
+     * Entry to validate user information.
+     */
+    private final transient PsBasic.Entry entry;
+
+    /**
+     * Ctor.
+     * @param basic Entry
+     */
+    public PsBasic(final PsBasic.Entry basic) {
+        this.entry = basic;
+    }
+
+    @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    public Opt<Identity> enter(final Request request) throws IOException {
+        Opt<BasicAuth> auth = new Opt.Empty<PsBasic.BasicAuth>();
+        Opt<Identity> identity = new Opt.Empty<Identity>();
+        for (final String head : request.head()) {
+            if (head.startsWith(AUTH_HEAD)) {
+                auth = this.readAuthContentOnHead(head);
+                break;
+            }
+        }
+        if (auth.has() && this.entry.check(
+            auth.get().getUser(),
+            auth.get().getPass()
+        )) {
+            final ConcurrentMap<String, String> props =
+                new ConcurrentHashMap<String, String>(0);
+            identity = new Opt.Single<Identity>(
+                new Identity.Simple(
+                    String.format("urn:basic:%s", auth.get().getUser()),
+                    props
+                )
+            );
+        }
+        return identity;
+    }
+
+    @Override
+    public Response exit(final Response response, final Identity identity)
+        throws IOException {
+        return response;
+    }
+
+    /**
+     * Read authentication content that is received on the head.
+     * @param head Head
+     * @return BasicAuth instance.
+     */
+    private Opt<BasicAuth> readAuthContentOnHead(final String head) {
+        final String authorization = new String(
+            DatatypeConverter.parseBase64Binary(
+                head.split(AUTH_HEAD)[1].trim()
+            )
+        );
+        final String user = authorization.split(":")[0];
+        return new Opt.Single<PsBasic.BasicAuth>(
+            new BasicAuth(
+                user,
+                authorization.substring(user.length() + 1)
+            )
+        );
+    }
+
+    /**
+     * Entry interface that is used to check if the received information is
+     * valid.
+     *
+     * @author Endrigo Antonini (teamed@endrigo.com.br)
+     * @version $Id$
+     * @since 0.20
+     */
+    public interface Entry {
+
+        /**
+         * Check if is a valid user.
+         * @param user User
+         * @param pwd Password
+         * @return If valid it return <code>true</code>.
+         */
+        boolean check(String user, String pwd);
```

Comment:
@antonini looks like it's better to return `Opt<Identity>` here and let our users decide what is the identity, with all its attributes, etc.